apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetService.java

```diff
@@ -171,10 +172,18 @@ public Dataset findById(@NonNull UUID id) {
                 .toStream()
                 .collect(toMap(ExperimentSummary::datasetId, Function.identity()));
 
+        Map<UUID, DatasetItemSummary> datasetItemSummaryMap = datasetItemDAO
+                .findDatasetItemSummaryByDatasetIds(List.of(dataset.id()))
+                .contextWrite(ctx -> AsyncUtils.setRequestContext(ctx, requestContext))
+                .toStream()
+                .collect(toMap(DatasetItemSummary::datasetId, Function.identity()));
+
         var summary = experimentSummary.computeIfAbsent(dataset.id(), ExperimentSummary::empty);
+        var datasetItemSummary = datasetItemSummaryMap.computeIfAbsent(dataset.id(), DatasetItemSummary::empty);
```

Comment:
Minor: this is copied and pasted multiple times (it was already, you're just following the convention). It'd be nice to encapsulate it into smaller private aux methods.

Reply:
Done